code/game/objects/machinery/vending/vending.dm

```diff
@@ -582,8 +649,9 @@ GLOBAL_LIST_INIT(vending_white_items, typecacheof(list(
 
 /obj/machinery/vending/proc/stock(obj/item/item_to_stock, mob/user, recharge = FALSE)
 	//More accurate comparison between absolute paths.
-	for(var/iter in (product_records + hidden_records + coin_records))
-		var/datum/data/vending_product/R = iter //Let's try with a new datum.
+	var/list/locallist = list()
+	locallist.Insert(product_records + hidden_records + coin_records) // this supposedly adds just the datums on a locallist so it aint nested in assocs.
+	for(var/datum/vending_product/R AS in locallist)
```

Comment:
Explain this, why are you using insert on a local var that just caches this

Reply:
i just confused myself in this probably